config/nrfconnect/overlay-sample-defaults.conf

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+CONFIG_NCS_SAMPLES_DEFAULTS=y
+
+CONFIG_CPLUSPLUS=y
+CONFIG_LIB_CPLUSPLUS=y
+CONFIG_NEWLIB_LIBC=y
+
+# Export POSIX names for functions implementing a subset of POSIX standard in Zephyr
+CONFIG_POSIX_API=y
+CONFIG_PTHREAD_IPC=y
+CONFIG_EVENTFD=y
+
+CONFIG_CHIP=y
+
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=n
+
+# Application stack size
+CONFIG_MAIN_STACK_SIZE=8192
+CONFIG_INIT_STACKS=y
+
+# # Disable certain parts of Zephyr IPv6 stack
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
+CONFIG_NET_PKT_TX_COUNT=16
+CONFIG_NET_BUF_RX_COUNT=80
+CONFIG_NET_BUF_TX_COUNT=80
+
+# Bluetooth Low Energy configs
+CONFIG_BT=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_DEVICE_APPEARANCE=0
+CONFIG_BT_DEVICE_NAME_DYNAMIC=y
+CONFIG_BT_DEVICE_NAME_MAX=15
+CONFIG_BT_MAX_CONN=2
+
+# L2 OpenThread enabling
+CONFIG_NET_L2_OPENTHREAD=y
+
+# OpenThread configs
+CONFIG_OPENTHREAD_DHCP6_SERVER=y
+CONFIG_OPENTHREAD_COMMISSIONER=y
```

Comment:
I would say we don't need the `COMMISSIONER`, `BORDER_AGENT`, `BORDER_ROUTER` and `UDP_FORWARD` features in CHIP.

If they are not enabled by default, we should enable: `DHCP6_CLIENT`, `MTD_NETDIAG`, `SLAAC` for sure. I would also add support for DNS and SNTP: `DNS_CLIENT`, `SNTP_CLIENT` as a  bonus.

Btw. how hard would be to use pre-compiled libraries? Can we have an overlay for it as well? (not in this PR).

Reply:
Done. I'm not sure, but I suspect that it should be possible. I will ask Marek Porwisz about this, as I think that he was doing something similar in the Zephyr.